metrics/fidelity/fidelity.py

```diff
@@ -0,0 +1,145 @@
+import logging
+
+import numpy as np
+
+import metrics.fidelity.utility as utility
+import datasets
+
+
+logger = logging.getLogger(__name__)
+
+_CITATION = """
+citation is under construction 
+@misc{carton2020evaluating,
+      title={Evaluating and Characterizing Human Rationales}, 
+      author={Samuel Carton and Anirudh Rathore and Chenhao Tan},
+      year={2020},
+      eprint={2010.04736},
+      archivePrefix={arXiv},
+      primaryClass={cs.CL}
+}
+"""
+
+_DESCRIPTION = """
+description is under construction
```

Comment:
you should drop these under construction sentences?

Reply:
Description should be something like: 
"This metric computes fidelity (Yu et al. 2019, DeYoung et al. 2019) and normalized fidelity (Carton et al. 2020). 

Fidelity is a measure of rationale faithfulness, measuring whether the information contained in an extractive rationale is sufficient or necessary (aka comprehensive) for the model to make similar predictions as it does with full information. A good rationale should be highly sufficient, and optionally highly comprehensive depending on the specific use case. 

Normalization is an optional addition which adjusts fidelity scores to account for baseline model behavior. This is helpful when comparing scores across different models and datasets. See (Carton et al. 2020) for further details.

References:
Samuel Carton, Anirudh Rathore, and Chenhao Tan. 2020. Evaluating and Characterizing Human Rationales. In Proceedings of the 2020 Conference on Empirical Methods in Natural Language Processing.
Jay DeYoung, Sarthak Jain, Nazneen Fatema Rajani, Eric Lehman, Caiming Xiong, Richard Socher, and Byron C. Wallace. 2019. ERASER: A Benchmark to Evaluate Rationalized NLP Models. arXiv preprint, November. arXiv: 1911.03429.
Mo Yu, Shiyu Chang, Yang Zhang, and Tommi S. Jaakkola. 2019. Rethinking Cooperative Rationalization: Introspective Extraction and Complement Control. In Proceedings of the 2019 Conference on Empirical Methods in Natural Language Processing.